### PR TITLE
VPC ghost tags exception with retry

### PR DESCRIPTION
**Pull request by nasark**
https://github.com/ManageIQ/manageiq-providers-ibm_cloud/issues/285

@miq-bot add_label enhancement

### CODE REVIEW

**Comment by nasark** [comment]
@miq-bot assign @agrare

File: app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb [hunk 1]
```diff
@@ -115,7 +115,15 @@ def volume_profiles
   end
 
   def tags_by_crn(crn)
-    vpc.cloudtools.tagging.collection(:list_tags, :attached_to => crn, :providers => ["ghost"]).to_a
+    retried = false
+    begin
+      vpc.cloudtools.tagging.collection(:list_tags, :attached_to => crn, :providers => ["ghost"]).to_a
+    rescue IBMCloudSdkCore::ApiException
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb:121]: What was the exception when we hit a timeout?  I think we should check that the exception was the api request limit error versus something like a 403 forbidden

File: app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb [hunk 2]
```diff
@@ -115,7 +115,15 @@ def volume_profiles
...
+      raise if retried
+      sleep(5)
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb:123]: Minor but rubocop is right
```suggestion
      raise if retried
      
      sleep(5)
```

File: app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb (hunk 1 above)
Comment (nasark) [reply app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb:121]:
```
[----] E, [2022-12-02T15:52:13.534524#7:977c]   ERR -- evm: MIQ(ManageIQ::Providers::IbmCloud::VPC::CloudManager::Refresher#refresh) EMS: [IBM-CVPC-auto-created], id: [6] Refresh failed
[----] E, [2022-12-02T15:52:13.534822#7:977c]   ERR -- evm: [IBMCloudSdkCore::ApiException]: Error: , Code: , Information: You must wait 35068 ms before you can make ghost-tags get api requests with attached_to query parameter, X-global-transaction-id: gst-bfecd060-c20c-44af-96a7-46bd53b8425f  Method:[block (2 levels) in <class:LogProxy>]
```

Comment (agrare) [reply app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb:121]: Okay so we could do `err.message.match?(/You must wait \d+ ms before you can make ghost-tags get api requests/)`

**Comment by Fryguy** [comment]
Backported to `oparin` in commit 9189f533bd3a962bb4ae1f38f27a4135cfdea045.

```text
commit 9189f533bd3a962bb4ae1f38f27a4135cfdea045
Author: Adam Grare <adam@grare.com>
Date:   Fri Dec 16 16:02:23 2022 -0500

    Merge pull request #441 from nasark/exception_and_retry_ghost_tags
    
    VPC ghost tags exception with retry
    
    (cherry picked from commit 25e4ec76e215ca225b9b01799da4c4bf78c8da76)
```

**Comment by Fryguy** [comment]
The original issue was opened against morphy, so tagging for backport back to morphy.

**Pull request merged**